crates/accelerate/src/sabre_layout.rs

```diff
@@ -114,12 +123,33 @@ fn layout_trial(
     max_iterations: usize,
     num_swap_trials: usize,
     run_swap_in_parallel: bool,
+    starting_layout: &Vec<Option<usize>>,
 ) -> ([NLayout; 2], SabreResult) {
     // Pick a random initial layout and fully populate ancillas in that layout too
     let num_physical_qubits = distance_matrix.shape()[0];
     let mut rng = Pcg64Mcg::seed_from_u64(seed);
-    let mut physical_qubits: Vec<usize> = (0..num_physical_qubits).collect();
-    physical_qubits.shuffle(&mut rng);
+    let physical_qubits = if !starting_layout.is_empty() {
+        let used_bits: HashSet<usize> = starting_layout
+            .iter()
+            .filter_map(|x| x.as_ref())
+            .copied()
+            .collect();
+        let mut free_bits: Vec<usize> = (0..num_physical_qubits)
+            .filter(|x| !used_bits.contains(x))
+            .collect();
+        free_bits.shuffle(&mut rng);
+        (0..num_physical_qubits)
+            .map(|x| match starting_layout.get(x) {
+                Some(phys) => phys.unwrap_or_else(|| free_bits.pop().unwrap()),
+                None => free_bits.pop().unwrap(),
+            })
+            .collect()
+    } else {
+        let mut physical_qubits: Vec<usize> = (0..num_physical_qubits).collect();
+        physical_qubits.shuffle(&mut rng);
+        physical_qubits
+    };
+
```

Comment:
Yeah we don't need it, it would work fine without the if statement. But, I figured it was a bit more efficient to leave it like this. That being said I didn't measure it so I don't actually know how much overhead having a branch here vs doing creating 3 objects (one empty) and doing 2 iterations instead of one has. I can measure it to see if it's worth the extra code or not